release/src/router/others/entware-setup.sh

```diff
@@ -0,0 +1,130 @@
+#!/bin/sh
+
+BOLD="\033[1m"
+NORM="\033[0m"
+INFO="$BOLD Info: $NORM"
+ERROR="$BOLD *** Error: $NORM"
+WARNING="$BOLD * Warning: $NORM"
+INPUT="$BOLD => $NORM"
+
+i=1 # Will count available partitions (+ 1)
+cd /tmp
+
+echo -e "$INFO This script will guide you through the Entware installation."
+echo -e "$INFO Script modifies \"entware\" folder only on the chosen drive,"
+echo -e "$INFO no other data will be changed. Existing installation will be"
+echo -e "$INFO replaced with this one. Also some start scripts will be installed,"
+echo -e "$INFO the old ones will be saved on Entware partition with name"
+echo -e "$INFO like /tmp/mnt/sda1/jffs_scripts_backup.tgz"
+echo
+
+if [ ! -d /jffs/scripts ] ; then
+  echo -e "$ERROR Please \"Enable JFFS partition\" from \"Administration > System\""
+  echo -e "$ERROR web UI page at www.asusrouter.com/Advanced_System_Content.asp"
+  echo -e "$ERROR then reboot router and try again. Exiting..."
+  exit 1
+fi
+
+case $(uname -m) in
+  arm*)
```

Comment:
I recommend looking for "armv7l" instead, in case a future router switches to armv8 (it will probably happen within a few years) or armv7b (quite unlikely to happen however).